Assets/Editor Toolbox/Editor/ToolboxEditorSceneView.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+namespace Toolbox.Editor
+{
+    using Toolbox.Editor.SceneView;
+
+    [InitializeOnLoad]
+    public static class ToolboxEditorSceneView
+    {
+        static ToolboxEditorSceneView()
+        {
+            UpdateEventCallback();
+        }
+
+        private static List<GameObject> GetObjectsUnderCursor()
+        {
+            List<GameObject> hitObjects = new List<GameObject>();
+            GameObject[] hitObjectsArray;
+
+            int maxIterations = 50;
+            for (int i = 0; i < maxIterations; i++)
+            {
+                hitObjectsArray = hitObjects.ToArray();
+
+                GameObject go = HandleUtility.PickGameObject(Event.current.mousePosition, true, hitObjectsArray);
+                if (go == null)
+                {
+                    break;
+                }
+
+                hitObjects.Add(go);
+            }
+
+            return hitObjects;
+        }
+
+        private static void UpdateEventCallback()
+        {
+            if (UseToolboxSceneView && !IsEventAttached)
+            {
+                UnityEditor.SceneView.duringSceneGui += SceneView_DuringSceneGUI;
```

Comment:
Wouldn't it be better to:
```
UnityEditor.SceneView.duringSceneGui -= SceneView_DuringSceneGUI
UnityEditor.SceneView.duringSceneGui += SceneView_DuringSceneGUI
```
and ignore functionality in the callback if UseToolboxSceneView == false?